core/src/main/scala/org/apache/spark/util/collection/OpenHashSet.scala

```diff
@@ -223,6 +224,8 @@ class OpenHashSet[@specialized(Long, Int) T: ClassTag](
    */
   private def rehash(k: T, allocateFunc: (Int) => Unit, moveFunc: (Int, Int) => Unit) {
     val newCapacity = _capacity * 2
+    require(newCapacity <= OpenHashSet.MAX_CAPACITY,
```

Comment:
2^31 is positive, but it is not representable as a 32-bit signed int.

```
scala> val _capacity = 1 << 30
_capacity: Int = 1073741824

scala> val newCapacity = _capacity * 2
newCapacity: Int = -2147483648
```

So that's an important check.